app/controllers/api/v1/icons_controller.rb

```diff
@@ -35,24 +35,25 @@ def raw_icon
         head :no_content
       end
 
-      def override_icon
-        overriden_icon = Catalog::OverrideIcon.new(params.require(:icon_id), params.require(:portfolio_item_id))
-        render :json => overriden_icon.process.icon
-      end
-
       private
 
       def icon_params
-        params.require([:content, :portfolio_item_id])
+        params.require(:content)
         icon_patch_params
       end
 
       def icon_patch_params
-        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :id)
+        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :portfolio_id, :id)
```

Comment:
@lindgrenj6 can we use the ParameterMixin here to get the list of params, it should be picking it up from the openapi.json

Reply:
@mkanoor yes we can use it for `PortfolioItem` (updated that already) but not for Icon. Mainly due to the fact that we can't validate against the Icon schema since it has generic references (iconable) and we have to allow portfolio_id/portfolio_item_id to be passed in and handled appropriately.